src/store/index.ts

```diff
@@ -0,0 +1,750 @@
+    ipc.on("CONNECTED", (event, data) => {
+      store.commit("setUser", data);
+    });
+
+    ipc.on("USER_EXTRA", (event, data) => {
+      store.commit("setUserExtra", data);
+    });
+
+    ipc.on("SAVED_USER", (event, data) => {
+      store.commit("setSavedUserName", data);
+    });
+    /*
+        ipc.on("DEBUG_EVT", (event, data) => {
+            // eslint-disable-next-line no-console
+            console.log(data)
+        })
+        */
+
+    ipc.on("UPDATE_DOWNLOADED", () => {
+      store.dispatch("errDialog", {
+        title: "Update available",
+        body: "App finished downloading an update and will apply the next time you launch.",
+        canClose: false,
+      });
+    });
+
+    ipc.on("CONNECTION_LOST", () => {
+      store.dispatch("errDialog", {
+        title: "Disconnected from server",
+        body: "",
+        canClose: false,
+      });
+    });
+
+    ipc.on("sRoomUpdated", (event, data) => {
+      store.commit("roomUpdated", data);
+    });
+
+    ipc.on("sRoomFlagsUpdated", (event, data) => {
+      store.commit("roomFlagsUpdated", data);
+    });
+
+    ipc.on("sRoomDeleted", (event, data) => {
+      store.commit("resetRoom", data);
+    });
+
+    ipc.on("sUserJoinedRoom", (event, data) => {
+      store.commit("userJoinedRoom", data);
+    });
+
+    ipc.on("sUserLeftRoom", (event, data) => {
+      //store.commit("chatMsg", `[System] ${store.state.rooms.users[data.userId]} has left the room.`)
+      if (data.userId == store.state.user.id) {
+        store.commit("resetRoom");
+      } else {
+        store.commit("userLeftRoom", data);
+      }
+    });
+
+    ipc.on("sUserKicked", (event, data) => {
+      //store.commit("chatMsg", `[System] ${store.state.rooms.users[data.userId]} has been kicked.`)
+      if (data.userId == store.state.user.id) {
+        store.commit("resetRoom");
+      } else {
+        store.commit("userLeftRoom", data);
+      }
+    });
+
+    ipc.on("sUserBanned", (event, data) => {
+      //store.commit("chatMsg", `[System] ${store.state.rooms.users[data.userId]} has been banned.`)
+      if (data.userId == store.state.user.id) {
+        store.commit("resetRoom");
+      } else {
+        store.commit("userLeftRoom", data);
+      }
+    });
+
+    ipc.on("sUserReadyState", (event, data) => {
+      //console.log({ gotready: data })
+      store.commit("userReadyState", data);
+    });
+
+    ipc.on("sRoomAddToList", (event, data) => {
+      store.commit("addRoom", data.room);
+    });
+
+    ipc.on("sRoomDeleted", (event, data) => {
+      store.commit("deleteRoom", data.id);
+    });
+
+    ipc.on("sRoomList", (event, data) => {
+      store.commit("setRooms", data.rooms);
+    });
+
+    ipc.on("sChat", (event, data) => {
+      store.commit("pushChat", data);
+    });
+
+    /*
+        ipc.on("sDisconnected", (e, reason) => {
+            //Show disconnection msg ?
+        })
+        */
+  };
+};
+
+const useStore = defineStore("store", () => {
```

Comment:
The store now looks way more like those new Vue components do.

1. The state is just a `reactive` (and `ref`s can also be used)
2. Getters are `computed()`
3. Mutations are plain old functions, and are an obsolete concept
4. Actions are plain old functions